### PR TITLE
Space heater little buff + smol refactor and docs

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Increased base power from 20000 to 40000, is still pretty slow at t1, but not painful anymore, every tier power has been doubled as effect
Refactored the construction of the spess heater to a more sane one (no more create cell and delete it afterwards), now the players will create a subtype of the heater already without the cell, while the main one is the admin and map spawned one with the default cell
Refactored attackby to use default procs and early returns
Variables now uses snake cases

## Why It's Good For The Game
Better code and docs, buff to a stupidly slow machine to help fix the atmosphere better

## Changelog
:cl:
balance: buffed space heater to have double power (from 20000 to 40000 at t1), every tier has been doubled in power as effect
refactor: refactored construction code to use subtype instead of deleting old cell on construction, changed attackby to use default procs and early returns
code: added documentation and variables with snake cases
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
> balance: buffed t1 space heater to have double power (from 20000 to 40000)

might wanna change that cause it also doubles the power of upgraded space heaters

**Comment by gryffinp** [comment]
Please for the love of god merge this.

File: code/game/machinery/spaceheater.dm [hunk 1]
```diff
@@ -143,32 +145,34 @@
 		cell.emp_act(severity)
 
 /obj/machinery/space_heater/attackby(obj/item/I, mob/user, params)
+	. = ..()
```
Comment (LemonInTheDark) [inline code/game/machinery/spaceheater.dm:148]: Should we really be doing this? don't the parents of attackby do things like try and hit the machine with the tool?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/game/machinery/spaceheater.dm [hunk 2]
```diff
@@ -15,18 +15,28 @@
 	max_integrity = 250
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, RAD = 100, FIRE = 80, ACID = 10)
 	circuit = /obj/item/circuitboard/machine/space_heater
-	/// We don't use area power, we always use the cell
+	//We don't use area power, we always use the cell
```
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/game/machinery/spaceheater.dm:18]: Is not a new var, no need for triple /

File: code/game/machinery/spaceheater.dm [hunk 3]
```diff
@@ -58,12 +61,11 @@
...
 		. += "There is no power cell installed."
 	if(in_range(user, src) || isobserver(user))
-		. += "<span class='notice'>The status display reads: Temperature range at <b>[settableTemperatureRange]°C</b>.<br>Heating power at <b>[siunit(heatingPower, "W", 1)]</b>.<br>Power consumption at <b>[(efficiency*-0.0025)+150]%</b>.</span>" //100%, 75%, 50%, 25%
+		. += "<span class='notice'>The status display reads: Temperature range at <b>[settable_temperature_range]°C</b>.<br>Heating power at <b>[siunit(heating_power, "W", 1)]</b>.<br>Power consumption at <b>[(efficiency*-0.0025)+150]%</b>.</span>" //100%, 75%, 50%, 25%
 
 /obj/machinery/space_heater/update_icon_state()
-	icon_state = "[base_icon_state]-[on ? mode : "off"]"
 	. = ..()
-	return
+	icon_state = "[base_icon_state]-[on ? mode : "off"]"
```
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/game/machinery/spaceheater.dm:68]: No reason to call parent after setting the icon state and then return, just putting this up to date with guidelines

File: code/game/machinery/spaceheater.dm (hunk 1 above)
Comment (Ghilker) [reply code/game/machinery/spaceheater.dm:148]: In theory yes, it attacks the object with the item you are wielding, but if you return TRUE the obj won't be hit, is all tested and working as it should

Comment (LemonInTheDark) [reply code/game/machinery/spaceheater.dm:148]: I'll hold you to that

**Pull request merged**